src/box/iproto.cc

```diff
@@ -568,11 +730,18 @@ iproto_check_msg_max(struct iproto_thread *iproto_thread)
 	return request_count > (size_t) iproto_msg_max;
 }
 
+static inline void
+iproto_msg_free(struct iproto_thread *iproto_thread, struct iproto_msg *msg)
+{
+	mempool_free(&iproto_thread->iproto_msg_pool, msg);
+	check_no_alive_requests(iproto_thread);
```

Comment:
You call this function which push message to cbus! every time when you delete iproto message. It's doubles message in cbus queue. I think it's very bad for performance and latency.

Reply:
It pushes message to cbus only if it is last message in thread